var/spack/repos/builtin/packages/rstudio/package.py

```diff
@@ -0,0 +1,124 @@
+    depends_on('java@8:')
+
+    depends_on('r-base64enc', when='+notebook')
+    depends_on('r-digest', when='+notebook')
+    depends_on('r-evaluate', when='+notebook')
+    depends_on('r-glue', when='+notebook')
+    depends_on('r-highr', when='+notebook')
+    depends_on('r-htmltools', when='+notebook')
+    depends_on('r-jsonlite', when='+notebook')
+    depends_on('r-knitr', when='+notebook')
+    depends_on('r-magrittr', when='+notebook')
+    depends_on('r-markdown', when='+notebook')
+    depends_on('r-mime', when='+notebook')
+    depends_on('r-rmarkdown', when='+notebook')
+    depends_on('r-stringi', when='+notebook')
+    depends_on('r-stringr', when='+notebook')
+    depends_on('r-tinytex', when='+notebook')
+    depends_on('r-xfun', when='+notebook')
+    depends_on('r-yaml', when='+notebook')
+
+    # to use node-js provided by spack
+    patch('https://src.fedoraproject.org/rpms/rstudio/raw/5bda2e290c9e72305582f2011040938d3e356906/f/0004-use-system-node.patch',
+          sha256='4a6aff2b586ddfceb7c59215e5f4a03f25b08fcc55687acaa6ae23c11d75d0e8')
+
+    def cmake_args(self):
+        args = [
+            '-DRSTUDIO_TARGET=Desktop',
+            '-DRSTUDIO_PACKAGE_BUILD=Yes',
+            '-DRSTUDIO_USE_SYSTEM_YAML_CPP=Yes',
+            '-DRSTUDIO_USE_SYSTEM_BOOST=Yes',
+            '-DRSTUDIO_USE_SYSTEM_SOCI=Yes',
+            '-DQT_QMAKE_EXECUTABLE="{0}"'.format(self.spec['qt'].prefix.bin.qmake),
+        ]
+
+        return args
+
+    def setup_build_environment(self, env):
+        env.set('RSTUDIO_TOOLS_ROOT', self.prefix.tools)
+#        env.set('JAVA_HOME', self.spec['java'].prefix)
+
+    def patch(self):
+        # fix hardcoded path for node-js in use_system_node patch
+        filter_file('<property name="node.bin" value="/usr/bin/node"/>',
+                    '<property name="node.bin" value="{0}"/>'.format(
+                        self.spec['node-js'].prefix.bin.node),
+                    'src/gwt/build.xml',
+                    string=True)
+
+        # remove hardcoded soci path to use spack soci
+        if self.spec['soci'].version <= Version('4.0.0'):
+            soci_lib = self.spec['soci'].prefix.lib64
+        else:
+            soci_lib = self.spec['soci'].prefix.lib
+        filter_file('set(SOCI_LIBRARY_DIR "/usr/lib")',
+                    'set(SOCI_LIBRARY_DIR "{0}")'.format(soci_lib),
+                    'src/cpp/CMakeLists.txt',
+                    string=True)
+
+        # unbundle icu libraries
+        filter_file('${QT_LIBRARY_DIR}/${ICU_LIBRARY}.so',
+                    join_path(self.spec['icu4c'].prefix.lib, '${ICU_LIBRARY}.so'),
+                    'src/cpp/desktop/CMakeLists.txt',
+                    string=True)
+
+    @run_before('cmake')
+    def install_deps(self):
+        deps = Executable('./dependencies/common/install-dictionaries')
+        deps()
+        deps = Executable('./dependencies/common/install-mathjax')
+        deps()
```

Comment:
Are we vendoring dependencies here? In case, is there an easy way to avoid that?

Reply:
Unfortunately yes. I did not find an easy way to avoid it at the time and since it only where text files and no binaries that was the easiest was.
I can take another look tomorrow since especially for the `install-dictionaries` I might have found another solution.

Reply:
Ok, then I'll wwait for that. Ping me when you're ready to merge this.

Reply:
Sorry but what I tried did not work out. 

The `install-dictionaries` dependency are just some dictionaries for spellchecking and `install-mathjax` are some fonts.  Since there are no binaries involved here just text files would it be ok, to leave it as is for now?